shared-lib/azure/azure-http/src/main/java/org/opennms/horizon/shared/azure/http/AzureHttpClient.java

```diff
@@ -51,12 +53,39 @@
 import java.nio.charset.StandardCharsets;
 import java.time.Duration;
 import java.time.temporal.ChronoUnit;
-import java.util.LinkedList;
-import java.util.List;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
 
 @Slf4j
 public class AzureHttpClient {
+    public enum ResourcesType {
+        PUBLIC_IP_ADDRESSES("publicIPAddresses"), NETWORK_INTERFACES("networkInterfaces"), NODE("node");
+        private final String metricName;
+        private static final Map<String, ResourcesType> metricNameMap = new HashMap<>();
+
+        static {
+            for (ResourcesType value : ResourcesType.values()) {
+                metricNameMap.put(value.getMetricName(), value);
+            }
+        }
+
+        ResourcesType(String metricName) {
+            this.metricName = metricName;
+        }
+
+        public String getMetricName() {
+            return metricName;
+        }
+
+        public static ResourcesType fromMetricName(String metricName) {
+            var type = metricNameMap.get(metricName);
+            if (type != null) {
+                return type;
+            }
+            throw new IllegalArgumentException();
```

Comment:
Unlikely to happen but we should have a message in the exception with the parameter to enhance traceability

```suggestion
            throw new IllegalArgumentException("Invalid value: " + metricName);
```